src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.NugetReference.cs

```diff
@@ -0,0 +1,154 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Packaging;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.AddImport
+{
+    internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSyntax>
+    {
+        private class PackageReference : Reference
+        {
+            private readonly IPackageInstallerService _installerService;
+            private readonly string _packageName;
+
+            public PackageReference(
+                AbstractAddImportCodeFixProvider<TSimpleNameSyntax> provider,
+                IPackageInstallerService installerService,
+                SearchResult searchResult,
+                string packageName)
+                : base(provider, searchResult)
+            {
+                _installerService = installerService;
+                _packageName = packageName;
+            }
+
+            public override Task<CodeAction> CreateCodeActionAsync(
+                Document document, SyntaxNode node, bool placeSystemNamespaceFirst, CancellationToken cancellationToken)
+            {
+                return Task.FromResult<CodeAction>(new PackageReferenceCodeAction(this, document, node, placeSystemNamespaceFirst));
+            }
+
+            private class PackageReferenceCodeAction : CodeAction
+            {
+                private readonly PackageReference _reference;
+                private readonly string _title;
+                private readonly ImmutableArray<CodeAction> _childCodeActions;
+
+                public override string Title => _title;
+
+                internal override bool HasCodeActions => true;
+
+                internal override ImmutableArray<CodeAction> GetCodeActions() => _childCodeActions;
+
+
+                public PackageReferenceCodeAction(
+                    PackageReference reference,
+                    Document document,
+                    SyntaxNode node,
+                    bool placeSystemNamespaceFirst) 
+                {
+                    _reference = reference;
+
+                    _title = $"{reference.provider.GetDescription(reference.SearchResult.NameParts)} ({string.Format(FeaturesResources.from_0, reference._packageName)})";
+
+                    var installedVersions = reference._installerService.GetInstalledVersions(reference._packageName);
+                    var versionsAndSplits = installedVersions.Select(v => new { Version = v, Split = v.Split('.') }).ToList();
+
+                    versionsAndSplits.Sort((v1, v2) =>
+                    {
+                        var diff = CompareSplit(v1.Split, v2.Split);
+                        return diff != 0 ? diff : -v1.Version.CompareTo(v2.Version);
+                    });
+
+                    var codeActions = new List<CodeAction>();
+
+                    // Add an action 
```

Comment:
Explain that we're adding actions for every local version + 1 action for the latest (as per nuget.org)